crates/forge/src/multi_runner.rs

```diff
@@ -147,32 +124,56 @@ impl MultiContractRunner {
             })
     }
 
-    /// Executes _all_ tests that match the given `filter`
+    /// Executes _all_ tests that match the given `filter`.
+    ///
+    /// The same as [`test`](Self::test), but returns the results instead of streaming them.
+    ///
+    /// Note that returns only when all tests have been executed.
+    pub async fn test_map(
+        &mut self,
+        filter: &dyn TestFilter,
+        test_options: TestOptions,
+    ) -> BTreeMap<String, SuiteResult> {
+        self.test_iter(filter, test_options).await.collect()
+    }
+
+    /// Executes _all_ tests that match the given `filter`.
+    ///
+    /// The same as [`test`](Self::test), but returns the results instead of streaming them.
+    ///
+    /// Note that returns only when all tests have been executed.
+    pub async fn test_iter(
+        &mut self,
+        filter: &dyn TestFilter,
+        test_options: TestOptions,
+    ) -> impl Iterator<Item = (String, SuiteResult)> {
+        let (tx, rx) = mpsc::channel();
+        self.test(filter, tx, test_options).await;
+        rx.into_iter()
+    }
+
+    /// Executes _all_ tests that match the given `filter`.
     ///
     /// This will create the runtime based on the configured `evm` ops and create the `Backend`
     /// before executing all contracts and their tests in _parallel_.
     ///
     /// Each Executor gets its own instance of the `Backend`.
     pub async fn test(
         &mut self,
-        filter: impl TestFilter,
-        stream_result: Option<Sender<(String, SuiteResult)>>,
+        filter: &dyn TestFilter,
+        stream_result: mpsc::Sender<(String, SuiteResult)>,
```

Comment:
What I also want to have here (for followups) is dedicated types instead of tuples.

dealing with `BtreeMap<String, Suiteresult>` is a bit horrible, I think we want a wrapper type so we can add useful helper functions